src/documents/views.py

```diff
@@ -1013,8 +1031,54 @@ def post(self, request, *args, **kwargs):
                 return HttpResponseForbidden("Insufficient permissions")
 
         try:
+            modified_field = self.MODIFIED_FIELD_BY_METHOD[method.__name__]
+            if settings.AUDIT_LOG_ENABLED and modified_field:
+                old_documents = list(
+                    Document.objects.filter(pk__in=documents).values(
+                        "pk",
+                        "correspondent",
+                        "document_type",
+                        "storage_path",
+                        "tags",
+                        "custom_fields",
+                        "deleted_at",
+                        "checksum",
+                    ),
+                )
+
             # TODO: parameter validation
             result = method(documents, **parameters)
+
+            if settings.AUDIT_LOG_ENABLED and modified_field:
+                new_documents = Document.objects.filter(pk__in=documents)
+                for doc in new_documents:
+                    old_value = next(
```

Comment:
I'm not sure how much we're concerned about performance, but I think this has to go through the list each loop iteration.

Maybe a dict like `old_documents = {obj.pk: obj for obj in queryset}`?

Reply:
Performance always a good goal, thanks!